hp/src/ErrorHandler.hs

```diff
@@ -0,0 +1,11 @@
+module ErrorHandler where
+
+import AbstractSyntaxTree (Identifier)
+
+data SemanticError
```

Comment:
Maybe you can add a type analyzer, but consider whether you're okay with the remaining time:
![2024-07-04_21-32](https://github.com/luiggymamani/hufflepuff-own-language/assets/135136456/90c5fd8c-3f86-4a54-9d48-089bd9c04823)